flexmeasures/api/dev/sensors.py

```diff
@@ -54,30 +66,76 @@ def get_chart(self, id: int, **kwargs):
         },
         location="query",
     )
-    def get_chart_data(self, id: int, **kwargs):
+    def get_chart_data(self, id: int, sensor: Sensor, **kwargs):
         """GET from /sensor/<id>/chart_data
 
         .. :quickref: Chart; Download time series for use in charts
 
         Data for use in charts (in case you have the chart specs already).
         """
-        sensor = get_sensor_or_abort(id)
         return sensor.search_beliefs(as_json=True, **kwargs)
 
-    def get(self, id: int):
+    @route("/<id>/chart_annotations/")
+    @use_kwargs(
+        {"sensor": SensorIdField(data_key="id")},
+        location="path",
+    )
+    @use_kwargs(
+        {
+            "event_starts_after": AwareDateTimeField(format="iso", required=False),
+            "event_ends_before": AwareDateTimeField(format="iso", required=False),
+            "beliefs_after": AwareDateTimeField(format="iso", required=False),
+            "beliefs_before": AwareDateTimeField(format="iso", required=False),
+        },
+        location="query",
+    )
+    def get_chart_annotations(self, id: int, sensor: Sensor, **kwargs):
+        """GET from /sensor/<id>/chart_annotations
+
+        .. :quickref: Chart; Download annotations for use in charts
+
+        Annotations for use in charts (in case you have the chart specs already).
+        """
+        df = sensor.generic_asset.search_annotations(
+            annotation_starts_after=kwargs.get("event_starts_after", None),
+            annotation_ends_before=kwargs.get("event_ends_before", None),
+            as_frame=True,
+        )
+
+        # Wrap on whitespace with some max line length
```

Comment:
Am I correct that this wrapping and stacking is logic only being done, but not via `/sensor/<id>/chart`->`Sensor.chart`? 

Is that something we can fix easily? It seems we could document this better, in any case.

Reply:
Good catch. I now applied this logic there, too.